core/domain/exp_services.py

```diff
@@ -513,27 +507,13 @@ def get_exploration_ids_matching_query(query_string, cursor=None):
         exp_ids, search_cursor = search_services.search_explorations(
             query_string, remaining_to_fetch, cursor=search_cursor)
 
-        invalid_exp_ids = []
-        for ind, model in enumerate(
+        for ind, _ in enumerate(
                 exp_models.ExpSummaryModel.get_multi(exp_ids)):
-            if model is not None:
-                returned_exploration_ids.append(exp_ids[ind])
-            else:
-                invalid_exp_ids.append(exp_ids[ind])
+            returned_exploration_ids.append(exp_ids[ind])
 
         if (len(returned_exploration_ids) == feconf.SEARCH_RESULTS_PAGE_SIZE
                 or search_cursor is None):
             break
-        else:
-            logging.error(
-                'Search index contains stale exploration ids: %s' %
-                ', '.join(invalid_exp_ids))
-
-    if (len(returned_exploration_ids) < feconf.SEARCH_RESULTS_PAGE_SIZE
```

Comment:
Why we are deleting this line?

Reply:
The returned exp ids will always be lesser or equal to SEARCH_RESULTS_PAGE_SIZE. If its lesser than SEARCH_RESULTS_PAGE_SIZE, the cursor will be None as there are no more page to sow and if its not none then it means there are more pages to show and in that case returned_exploration_ids will be greater than SEARCH_RESULTS_PAGE_SIZE

Reply:
I think this entirely depends on the decision of deleting the[ line 519](https://github.com/oppia/oppia/pull/6844/files/418a373c8b00d80c7adc4ee14a575dd0fcfff5e2#diff-6abb716589c61217f40bf3386182859cL519). So let's discuss this in the above comment.

Reply:
How does this depends on the line 519? It depends on returned_exploration_ids but the previous lines deal with invalid_exp_ids

Reply:
> How does this depends on the line 519? I

`len(returned_exploration_ids) < feconf.SEARCH_RESULTS_PAGE_SIZE` can be possible if the returned list of search_document doesn't correspond to one or more existing exploration model.

Reply:
> len(returned_exploration_ids) < feconf.SEARCH_RESULTS_PAGE_SIZE can be possible if the returned list of search_document doesn't correspond to one or more existing exploration model.

Yes but won't the search cursor be None then?

Reply:
Assume we have 50 documents with for the search query, for the 1st round we extracted 20 (as SEARCH_RESULTS_PAGE_SIZE  = 20)documents (the cursor will `xyz...`) out of which only 20 explorations exist.

In this case len(returned_exploration_ids) < feconf.SEARCH_RESULTS_PAGE_SIZE and the cursour won't be None.

Reply:
The `returned_exploration_ids` will be those exp ids for which there is an ExpSummaryModel(see line 520). So if total explorations > SEARCH_RESULTS_PAGE_SIZE, returned_exploration_ids will always be less(in this case cursor will be None) or equal to SEARCH_RESULTS_PAGE_SIZE. And if total explorations < SEARCH_RESULTS_PAGE_SIZE, the cursor will be None.

Reply:
> returned_exploration_ids will always be less(in this case cursor will be None) or equal to SEARCH_RESULTS_PAGE_SIZE

I agree that it would be either less or equal to SEARCH_RESULTS_PAGE_SIZE, but for less than a case, I don't think the cursor will always be None! (See my above comments)
> Assume we have 50 documents for a search query, in the 1st round we extracted 20 (as SEARCH_RESULTS_PAGE_SIZE = 20)documents (the cursor will xyz...) out of which only 19 explorations exist.
>
> In this case len(returned_exploration_ids) < feconf.SEARCH_RESULTS_PAGE_SIZE and the cursor won't be None, right?

Reply:
The above case will be possible if the returned_exploration_ids < feconf.SEARCH_RESULTS_PAGE_SIZE even after fetching MAX_ITERATIONS times.

Reply:
Done!